### PR TITLE
samples: shields: Update Nucleo shields to new code-sample extension

### PR DESCRIPTION
**Pull request by kartben**
Update all Nucleo shield samples and references to them so that they
use the new zephyr:code-sample extension.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

### CODE REVIEW

**Review by erwango — APPROVED** [review]
Thanks @kartben !

**Comment by kartben** [comment]
FWIW I am not sure things will scale all that well when we start migrating all the 60+ "sensor" samples, as they will all flag `sensor_interface` as a relevant API, since it will probably not be that helpful to list 60+ samples when browsing sensor API reference.

**Comment by jfischer-no** [comment]
> FWIW I am not sure things will scale all that well when we start migrating all the 60+ "sensor" samples, as they will all flag `sensor_interface` as a relevant API, and then it will probably not be that helpful anymore to show 60+ samples when browsing sensor API reference.

The hope from the past was that we would have samples per sensor type at some point and not sample for each sensor driver.

File: boards/shields/x_nucleo_iks01a3/doc/index.rst [hunk 1]
```diff
@@ -82,9 +82,9 @@ Programming
 
 Two samples are provided as examples for ``x-nucleo-iks01a3`` shield:
 
-- :ref:`x-nucleo-iks01a3-std-sample` application, to be used when the shield is configured
+- :zephyr:code-sample:`x-nucleo-iks01a3-std` sample application, to be used when the shield is configured
   in Standard Mode
-- :ref:`x-nucleo-iks01a3-shub-sample` application, to be used when the shield is configured
+- :zephyr:code-sample:`x-nucleo-iks01a3-shub` sample application, to be used when the shield is configured
```
Comment (jfischer-no) [inline boards/shields/x_nucleo_iks01a3/doc/index.rst:87]: nit: Could be reformatted as coding style line length also applies to documentation sources.

**Review by jfischer-no — APPROVED** [review]

File: boards/shields/x_nucleo_iks01a3/doc/index.rst (hunk 1 above)
Comment (kartben) [reply boards/shields/x_nucleo_iks01a3/doc/index.rst:87]: yes but coding style line length is 100 characters :)

Comment (jfischer-no) [reply boards/shields/x_nucleo_iks01a3/doc/index.rst:87]: And this line is 106.

Comment (kartben) [reply boards/shields/x_nucleo_iks01a3/doc/index.rst:87]: > And this line is 106.

touché :) was looking at the wrong diff. It's on my list to find a good RST linter and hook it up to CI...

**Pull request merged**